jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -1,274 +1,93 @@
+    See the :ref:`Unit Conversion Plugin Documentation <unit-conversion>` for more details.
+
+    Only the following attributes and methods are available through the
+    :ref:`public plugin API <plugin-apis>`:
+
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.show`
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.open_in_tray`
+    * ``spectral_unit`` (:class:`~jdaviz.core.template_mixin.SelectPluginComponent`):
+      Global unit to use for all spectral axes.
+    * ``flux_unit`` (:class:`~jdaviz.core.template_mixin.SelectPluginComponent`):
+      Global unit to use for all flux axes.
+    """
     template_file = __file__, "unit_conversion.vue"
 
-    current_flux_unit = Unicode().tag(sync=True)
-    current_spectral_axis_unit = Unicode().tag(sync=True)
-    new_flux_unit = Any().tag(sync=True)
-    new_spectral_axis_unit = Any().tag(sync=True)
-
-    spectral_axis_unit_equivalencies = List([]).tag(sync=True)
-    flux_unit_equivalencies = List([]).tag(sync=True)
+    spectral_unit_items = List().tag(sync=True)
+    spectral_unit_selected = Unicode().tag(sync=True)
+    flux_unit_items = List().tag(sync=True)
+    flux_unit_selected = Unicode().tag(sync=True)
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
-        self.hub.subscribe(self, SubsetCreateMessage, handler=self._on_viewer_subset_changed)
-        self.hub.subscribe(self, SubsetDeleteMessage, handler=self._on_viewer_subset_changed)
-
-        self._redshift = None
-        self.app.hub.subscribe(self, RedshiftMessage, handler=self._redshift_listener)
-
-        self._default_spectrum_viewer_reference_name = kwargs.get(
-            "spectrum_viewer_reference_name", "spectrum-viewer"
-        )
-
-        # when accessing the selected data, access the spectrum-viewer version
-        # TODO: we'll probably want to update unit-conversion to be able to act on cubes directly
-        # in the future
-        self.dataset._viewers = [self._default_spectrum_viewer_reference_name]
-        # require entries to be in spectrum-viewer (not other cubeviz images, etc)
-        self.dataset.add_filter('layer_in_spectrum_viewer')
-
-    def _on_viewer_subset_changed(self, *args):
-        if len(self.app.data_collection.subset_groups) == 0:
-            self.disabled_msg = ''
-        else:
-            self.disabled_msg = 'Please create Subsets only after unit conversion'
-
-    def _redshift_listener(self, msg):
-        '''Save new redshifts (including from the helper itself)'''
-        if msg.param == "redshift":
-            self._redshift = msg.value
-
-    @observe('dataset_selected')
-    def update_ui(self, event=None):
-        """
-        Set up UI to have all values of currently visible spectra.
-        """
-        spectrum = self.dataset.selected_obj
-        if spectrum is None:
+        # TODO: support for multiple viewers
+        # TODO: support for sky coordinate axes?
```

Comment:
We only use ICRS right now, so I am not sure if we need to worry about converting sky coordinates between different frames.
```suggestion
```